lms/djangoapps/course_api/serializers.py

```
@@ -96,6 +96,37 @@ def get_blocks_url(self, course_overview):
         return self.context['request'].build_absolute_uri(base_url)
 
 
+class CourseDetailMarketingSerializer(CourseSerializer):
+    """
+    Serializer for Course objects (like CourseDetailSerializer) but fetching some data from the marketing site.
+
+    This allows reuse of the marketing site data, as opposed to duplicating them in the LMS.
+    """
+
+    def get_data_with_marketing_overrides(self, original_serialized_data):
+        marketing_data = self.context.get('marketing_data')
+        if not marketing_data:
+            return original_serialized_data
+
+        overridden = {}
+        overridden.update(original_serialized_data)
+
+        overridden['effort'] = marketing_data['effort']
```

These are going to cost us a 500 error in case we decided to rename the field in the api, right?

---

Yes, and that's OK. It's better to fail explicitly rather than returning `None` suddenly!